eng/scripts/docs/Docs-ToC.ps1

```diff
@@ -0,0 +1,154 @@
+function Get-java-OnboardedDocsMsPackages($DocRepoLocation) {
+    $packageOnboardingFiles = "$DocRepoLocation/package.json"
+  
+    $onboardingSpec = ConvertFrom-Json (Get-Content $packageOnboardingFiles -Raw)
+    $allPackages = @{}
+    foreach ($spec in $onboardingSpec) {
+      $spec.packages | ForEach-Object {$allPackages[$_.packageArtifactId] = $null}
+    }
+    return $allPackages
+  }
+  
+function Get-java-DocsMsTocData($packageMetadata, $docRepoLocation) {
+    # Fallback to get package-level readme name if metadata file info does not exist
+    $packageLevelReadmeName = $packageMetadata.Package.Replace('azure-', '');
+
+    # If there is a metadata json for the package use the DocsMsReadmeName from
+    # the metadata function
+    if ($packageMetadata.PSObject.Members.Name -contains "FileMetadata") {
+        $readmeMetadata = &$GetDocsMsMetadataForPackageFn -PackageInfo $packageMetadata.FileMetadata
+        $packageLevelReadmeName = $readmeMetadata.DocsMsReadMeName
+    }
+
+    $packageTocHeader = $packageMetadata.Package
+    if ($packageMetadata.DisplayName) {
+        $packageTocHeader = $packageMetadata.DisplayName
+    }
+
+    $version = ""
+    $isPreview = $false
+    if($packageMetadata.VersionGA -ne "") {
+        $version = $packageMetadata.VersionGA
+    }
+    else {
+        $version = $packageMetadata.VersionPreview
+        $isPreview = $true
+    }
+    $children = Get-Toc-Children -package $packageMetadata.Package -groupId $packageMetadata.GroupId `
+        -version $version -docRepoLocation $docRepoLocation -isPreview $isPreview
```

Comment:
There's a ~potential for~ certainty of duplicates here, you'll need to use something like `Get-Unique` before returning.

Reply:
Updated suggestion to differentiate between `VersionPreview` and `VersionGA` in the calls to `Get-Toc-Children`

Reply:
This is loop into each of `$packageMetadata` 
The proposed changes are the same as original code.

Reply:
It is called in a loop but each `$clientPackage` in the loop is a row in the metadata CSV and can have both `VersionPreview` and `VersionGA` ... I probably should've named that variable `$metadataEntry` because `$clientPackage` implies it's a single entity :P